cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/metric/MetricsUtils.java

```diff
@@ -204,11 +204,15 @@ public static boolean isInterested(com.yammer.metrics.core.MetricName metricName
 
   /**
    * Convert a yammer metrics scope to a tags map.
+   * Skip parsing scopes that (1) are {@code null} or (2) have keys without a matching value
+   * (see https://github.com/linkedin/cruise-control/issues/1296).
    */
   private static Map<String, String> yammerMetricScopeToTags(String scope) {
     if (scope != null) {
       String[] kv = scope.split("\\.");
-      assert kv.length % 2 == 0;
+      if (kv.length % 2 != 0) {
+        return Collections.emptyMap();
```

Comment:
Do you know if it's safe to ignore generating the map for these metrics?

Reply:
This util function helps with the identification of whether given Kafka Yammer metrics are of interest to CC metrics reporter or not. The Kafka code ensures that `scope` is either a KV pair separated by a dot or `null` (see [relevant code](https://github.com/apache/kafka/blob/7e1c39f75abc443281f31f3c2f1e4829e738619d/core/src/main/scala/kafka/metrics/KafkaMetricsGroup.scala#L91)). Hence, unless these assumptions are broken, this should be safe.

However, in the unlikely scenario, where scope of Yammer metrics contains an odd number of elements, generating an empty map here may lead to a false positive -- i.e. the Yammer metric can be mistakenly identified as one of the metrics that CC metrics reporter is interested in. To prevent this scenario, I updated the handling logic for `kv.length % 2 != 0` case in `yammerMetricScopeToTags` to return `null` (to indicate `unacceptable` scope) and the callers of this util function to take a note of this response to avoid misidentification of tags.